tests/functional/api/test_objectstorage.py

```diff
@@ -1821,3 +1824,178 @@ def test_object_create_ext(self):
+        self.assertEqual(1, len(self.cache))
+
+    def test_object_locate(self):
+        properties = {'test1': '1', 'test2': '2'}
+        self.api.object_set_properties(
+            self.account, self.container, self.path, properties)
+        self.assertEqual(1, self.api.container._direct_request.call_count)
+        self.assertEqual(0, len(self.cache))
+
+        expected_obj_meta, expected_chunks = self.api.object_locate(
+            self.account, self.container, self.path, properties=False)
+        self.assertIsNotNone(expected_obj_meta)
+        self.assertIsNotNone(expected_chunks)
+        expected_obj_meta = expected_obj_meta.copy()
+        expected_chunks = copy.deepcopy(expected_chunks)
+        self.assertEqual(2, self.api.container._direct_request.call_count)
+        self.assertEqual(1, len(self.cache))
+
+        obj_meta, chunks = self.api.object_locate(
+            self.account, self.container, self.path, properties=False)
+        self.assertDictEqual(expected_obj_meta, obj_meta)
+        self.assertListEqual(expected_chunks, chunks)
+        self.assertEqual(2, self.api.container._direct_request.call_count)
+        self.assertEqual(1, len(self.cache))
+
+        expected_obj_meta['properties'] = properties
+        obj_meta, chunks = self.api.object_locate(
+            self.account, self.container, self.path, properties=True)
+        self.assertDictEqual(expected_obj_meta, obj_meta)
+        self.assertListEqual(expected_chunks, chunks)
+        self.assertEqual(3, self.api.container._direct_request.call_count)
+        self.assertEqual(1, len(self.cache))
+
+        obj_meta, chunks = self.api.object_locate(
+            self.account, self.container, self.path, properties=True)
+        self.assertDictEqual(expected_obj_meta, obj_meta)
+        self.assertListEqual(expected_chunks, chunks)
+        self.assertEqual(3, self.api.container._direct_request.call_count)
+        self.assertEqual(1, len(self.cache))
+
+        expected_obj_meta['properties'] = dict()
+        obj_meta, chunks = self.api.object_locate(
+            self.account, self.container, self.path, properties=False)
+        self.assertDictEqual(expected_obj_meta, obj_meta)
+        self.assertListEqual(expected_chunks, chunks)
+        self.assertEqual(3, self.api.container._direct_request.call_count)
+        self.assertEqual(1, len(self.cache))
+
+        expected_obj_meta['properties'] = properties
+        obj_meta = self.api.object_get_properties(
+            self.account, self.container, self.path)
+        self.assertDictEqual(expected_obj_meta, obj_meta)
+        self.assertEqual(3, self.api.container._direct_request.call_count)
+        self.assertEqual(1, len(self.cache))
+
+    def test_object_fetch(self):
+        expected_obj_meta, stream = self.api.object_fetch(
+            self.account, self.container, self.path)
+        self.assertIsNotNone(expected_obj_meta)
+        data = ''
```

Comment:
For forward compatibility, data buffers must be declared as bytes. This works with Python 2, and won't require any conversion for Python 3.